### PR TITLE
OKTA-296007: expose failure related events; AspNet and AspNetCore

### PR DESCRIPTION
**Pull request by bryanapellanes-okta**
## Issue \#
[OKTA-296007](https://oktainc.atlassian.net/browse/OKTA-296007)

## Code

- [x] Unit test(s)
- [x] Implementation

## Current behavior
Unhandled exceptions occur if authentication failures occur in Okta api.

## Desired behavior
Provide a way for developers to handle authentication failures

## Additional Context
https://github.com/okta/okta-aspnet/issues/120

### CODE REVIEW

File: Okta.AspNetCore/OktaMvcOptions.cs [hunk 1]
```diff
@@ -59,5 +60,15 @@ public class OktaMvcOptions : AspNet.Abstractions.OktaWebOptions
         /// </summary>
         /// <value>The OnUserInformationReceived event.</value>
         public Func<UserInformationReceivedContext, Task> OnUserInformationReceived { get; set; } = context => Task.CompletedTask;
+
+        /// <summary>
+        /// Gets or sets the event invoked when a failure occurs within the Okta api.
```
Comment (laura-rodriguez) [inline Okta.AspNetCore/OktaMvcOptions.cs:65]:
```suggestion
        /// Gets or sets the event invoked when a failure occurs within the Okta API.
```

File: Okta.AspNetCore/OktaMvcOptions.cs [hunk 2]
```diff
@@ -59,5 +60,15 @@ public class OktaMvcOptions : AspNet.Abstractions.OktaWebOptions
...
+        /// </summary>
+        public Func<RemoteFailureContext, Task> OnOktaException { get; set; } = context => Task.CompletedTask;
```
Comment (laura-rodriguez) [inline Okta.AspNetCore/OktaMvcOptions.cs:67]: What do you think if we rename this to `OnOktaApiFailure`? I'm wondering if using just `Okta` is too generic (the whole SDK can be considered Okta). Also, I was looking at the [existing events](https://docs.microsoft.com/ru-ru/dotnet/api/microsoft.aspnetcore.authentication.openidconnect.openidconnectevents?view=aspnetcore-1.0) and this seems to be consistent with the naming that's being used for others.

**Comment by bryanapellanes-okta** [comment]
@laura-rodriguez I'll update the README as part of this PR, along with your other recommendations.

**Comment by bryanapellanes-okta** [comment]
@laura-rodriguez for your reference: https://github.com/okta/okta-aspnet/pull/128/commits/19b9acb003fb7b96be9fb2ab60717fd69cf3e468

merging now.

**Pull request merged**